### PR TITLE
Conditionally delete stream

### PR DESCRIPTION
**Pull request by princemaple**
This adds the option to keep the stream when you `stop`, so you can re-`start`
@chris-rudmin what do you think?

### CODE REVIEW

**Comment by chris-rudmin** [comment]
Good idea. Things I would change:

* add option To global config with default false
* add a check for stream in initStream.
* add documentation in readme

**Comment by princemaple** [comment]
@chris-rudmin I've updated the PR, please take another look

File: recorder.js [hunk 1]
```diff
@@ -183,8 +189,16 @@ Recorder.prototype.stop = function(){
...
-    this.stream.stop();
-    delete this.stream;
+
+    if ( !this.config.reuseStream ) { this.clearStream(); }
+
     this.encoder.postMessage({ command: "done" });
   }
 };
+
+Recorder.prototype.clearStream = function() {
```
Comment (chris-rudmin) [inline recorder.js]: I like this new method :+1:

File: recorder.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ var Recorder = function( config ){
   this.config.encoderSampleRate = config.encoderSampleRate || 48000;
   this.config.encoderPath = config.encoderPath || 'oggopusEncoder.js';
   this.config.stream = config.stream || false;
+  this.config.reuseStream = !!config.reuseStream;
```
Comment (chris-rudmin) [inline recorder.js]: Could we use the convention other options use `config.leaveStreamOpen || false` I would also say `leaveStreamOpen` is more accurate than reuse stream.

File: README.md [hunk 3]
```diff
@@ -60,6 +61,10 @@ Creates a recorder instance.
 
 **stop** will cease capturing audio and disable the monitoring and mic input stream. Will request the recorded data and then terminate the worker once the final data has been published. Will raise the stop event when stopped.
 
+    rec.clearStream()
+
+**clearStream** will stop and delete the stream got from `initStream`, you will only ever call this manually if you have `config.reuseStream` set to true
```
Comment (chris-rudmin) [inline README.md]: What do you think of  `stopStream` instead of '`clearStream`

Comment (princemaple) [inline README.md]: IMO `stopStream` doesn't really cover the `delete this.stream` semantics

File: recorder.js (hunk 2 above)
Comment (princemaple) [inline recorder.js]: OK

**Comment by princemaple** [comment]
Thanks

**Comment by chris-rudmin** [comment]
Thanks for the great contribution @princemaple :+1:

**Pull request merged**